diff --git a/src/lread.c b/src/lread.c
index e1a4b33..4f17e39 100644
--- a/src/lread.c
+++ b/src/lread.c
@@ -467,7 +467,7 @@ readbyte_from_file (int c, Lisp_Object readcharfun)
     }
 
   block_input ();
-  c = getc (instream);
+  c = getc_unlocked (instream);
 
   /* Interrupted reads have been observed while reading over the network.  */
   while (c == EOF && ferror (instream) && errno == EINTR)
@@ -476,7 +476,7 @@ readbyte_from_file (int c, Lisp_Object readcharfun)
       QUIT;
       block_input ();
       clearerr (instream);
-      c = getc (instream);
+      c = getc_unlocked (instream);
     }
 
   unblock_input ();
